lambdas/static-asset-uploader/index.js

```diff
@@ -191,55 +105,99 @@ function createCatalogDirectory (staticBucketName) {
  * @param staticBucketName the name of the bucket the file should exist in
  * @returns {Promise.<Object>} the object returned by the headObject operation or the upload operation
  */
-function createSdkGenerationFile (staticBucketName) {
-  return exports.s3.headObject({ Bucket: staticBucketName, Key: 'sdkGeneration.json' }).promise()
-    .catch(_error => {
-      // assume it's a NotFound error, and upload a new version
-      console.log('Uploading sdkGeneration.json since it seems to not exist')
-      const params = { Bucket: staticBucketName, Key: 'sdkGeneration.json', Body: '{}' }
-      return exports.s3.upload(params).promise()
-    })
+async function createSdkGenerationFile (staticBucketName) {
+  try {
+    return await s3.headObject({ Bucket: staticBucketName, Key: 'sdkGeneration.json' }).promise()
+  } catch (_) {
+    // assume it's a NotFound error, and upload a new version
+    console.log('Uploading sdkGeneration.json since it seems to not exist')
+    const params = { Bucket: staticBucketName, Key: 'sdkGeneration.json', Body: '{}' }
+    return s3.upload(params).promise()
+  }
 }
 
-function addConfigFile (bucketName, event) {
-  const configObject = {
-    restApiId: event.ResourceProperties.RestApiId,
-    region: event.ResourceProperties.Region,
-    identityPoolId: event.ResourceProperties.IdentityPoolId,
-    userPoolId: event.ResourceProperties.UserPoolId,
-    userPoolClientId: event.ResourceProperties.UserPoolClientId,
-    userPoolDomain: event.ResourceProperties.UserPoolDomain,
-    // this is a stringified boolean here; let's go ahead and make it a real boolean
-    feedbackEnabled: event.ResourceProperties.FeedbackEnabled === 'true'
+class State {
```

Comment:
Why a class? (Also, make sure to run the tests. A lot of the mocking the tests were doing were based on mocking `exports.foo`.)

Reply:
I ran and updated the tests accordingly.

The reason I used a class was so I wasn't having to pass these two variables *everywhere* on each call. It's also easier to stub out - I can just stub out the methods without having to globally export them or restore their previous values.

Reply:
👍